### PR TITLE
Updated test

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-1026

### Change description ###
Fixed (hopefully) inconsistent failures of integration tests - instead of @RegisterExtension annotation started and stopped GreenMail explicitly before and after each test method. Run Jenkins build pipeline several times - never failed due to failures in the integration tests previously observed.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/services/email/EmailSenderTest.java [hunk 1]
```diff
@@ -29,20 +30,27 @@
     private static final String SUBJECT = "subject";
     private static final String RECIPIENT_1 = "Foo <foo@hmcts.net>";
     private static final String RECIPIENT_2 = "bar@hmcts.net";
-    private static final String[] RECIPIENTS = {RECIPIENT_1, RECIPIENT_2};
+    private static final String[] recipients = {RECIPIENT_1, RECIPIENT_2};
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/services/email/EmailSenderTest.java]: ?

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/services/email/EmailSenderTest.java [hunk 2]
```diff
@@ -37,12 +38,25 @@
     @Autowired
     private EmailSender emailSender;
 
-    @RegisterExtension
-    static GreenMailExtension greenMail = new GreenMailExtension(ServerSetupTest.SMTP);
```
Comment (doncem) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/services/email/EmailSenderTest.java]: this means we don't use greenmail extension and should be removed from project test sources

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/services/email/EmailSenderTest.java [hunk 3]
```diff
@@ -86,8 +100,8 @@ void should_handle_no_attachments() throws Exception {
             emptyMap()
         );
 
-        MimeMessageParser msg = new MimeMessageParser(greenMail.getReceivedMessages()[0]).parse();
-
+        assertThat(smtpServer.getReceivedMessages()).hasSize(RECIPIENTS.length);
+        MimeMessageParser msg = new MimeMessageParser(smtpServer.getReceivedMessages()[0]).parse();
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/services/email/EmailSenderTest.java]: Not sure this is the correct way but I found something on below lines. 

`smtpServer.getReceivedMessages()` will contains headers , body so extracting using util will make it clear imo.
```
  GreenMail mailServer = TestUtil.TestRuntimeModule.getMailServer();
    while(mailServer.getReceivedMessages().length < 1) {
      ThreadUtil.sleep(100);
    }
    String headers = GreenMailUtil.getHeaders(mailServer.getReceivedMessages()[0]);
    Assert.assertTrue(headers.contains("To: foo, bar"));
    Assert.assertTrue(headers.contains("Subject: StreamsSets Data Collector Alert - " +
        TestUtil.PIPELINE_TITLE_WITH_EMAIL + " - STOPPED"));
    Assert.assertTrue(headers.contains("From: sdc@localhost"));
    Assert.assertNotNull(GreenMailUtil.getBody(mailServer.getReceivedMessages()[0]));

```

**Comment by apereverzin** [comment]
PR-398

**Pull request closed without merging**